api/v1beta1/namespaced_resource.go

```diff
@@ -9,6 +9,11 @@ type NamespacedResource string
 
 type NamespacedResourceList []NamespacedResource
 
+// +kubebuilder:object:generate=false
+type NamespacedResourceImpl[T interface{}] interface {
+	Find(namespace string, name string) *T
```

Comment:
Since this is only used to compare with `nil`, we can refactor this to be `Exists(namespace string, name string)` which would get rid of the generics.

What do you think?